homeassistant/components/dreo/fan.py

```diff
@@ -0,0 +1,165 @@
+"""Support for Dreo fans."""
+from __future__ import annotations
+
+import logging
+import math
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage, int_states_in_range
+)
+from .const import DOMAIN, MANAGER, FAN_DEVICE, FAN_CONFIG
+from .device import DreoEntity
+from typing import Any
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.INFO)
+
+
+async def async_setup_entry(
+        hass: HomeAssistant,
+        config_entry: ConfigEntry,
+        async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Fan from a config entry."""
+
+    devices = hass.data[DOMAIN][config_entry.entry_id].get(FAN_DEVICE)
+
+    fans = [DreoFanHA(device.get("deviceName"), device, config_entry, device.get("deviceSn")) for device in devices]
+
+    async_add_entities(fans)
+
+
+class DreoFanHA(DreoEntity, FanEntity):
+    """Dreo fan."""
+
+    _attr_supported_features = (FanEntityFeature.PRESET_MODE
+                                | FanEntityFeature.SET_SPEED
+                                | FanEntityFeature.OSCILLATE)
+    _attr_name = None
```

Comment:
Name is set to None? But in the base entity (`DreoEntity`) the property is implemented. What is the right one?